treeherder/log_parser/utils.py

```diff
@@ -57,11 +48,16 @@ def _retry(e):
         # .retry() raises a RetryTaskError exception,
         # so nothing after this function will be executed
 
-    log_obj = expand_log_url(project, job_guid, job_log_url)
     log_description = "%s %s (%s)" % (project, job_guid, job_log_url)
-
     logger.debug("Downloading/parsing log for %s", log_description)
 
+    job_log = JobLog.objects.get(job__guid=job_guid,
+                                 url=job_log_url)
+
+    if job_log.status == JobLog.PARSED:
```

Comment:
`post_log_artifacts()` is only ever called by `parse_log()`, which is decorated with `@parser_task`.

`@parser_task` does it's own check of the parse status, so presumably this one is redundant?

In addition (albeit likely fodder for a separate refactor bug), seeing as `parser_task` already fetches the JobLog object, it's a shame that we have to do so again here.